SM-61 Stream interface

valerio8787 opened the pull request.

paulohenriquesg committed:
Added Unit tests to translator functions

paulohenriquesg committed:
More tests to the TranslatorFunctions

ievgenvarava committed:
Introduced tolerable exceptions.

valerio8787 committed:
SM-59: Added possibility to skip specified keys in each item of array

valerio8787 committed:
Merge pull request #11 from spryker-middleware/feature/SM-59-mapper-skip-item-keys

SM-59: Added possibility to skip specified keys in each item of array

ievgenvarava committed:
Merge branch 'dev' into feature/error-handling

ievgenvarava committed:
Merge pull request #10 from spryker-middleware/feature/error-handling

Introduced tolerable exceptions.

ievgenvarava committed:
Added map reading into abstract map class.

ievgenvarava committed:
Map file is passed as method parameter.

ievgenvarava committed:
Added file reader function to abstract dictionary.

ievgenvarava committed:
SM-62: Allowed translators to access values modified by previous translators in pipeline.

valerio8787 committed:
Merge pull request #14 from spryker-middleware/fix/SM-62-chained-translators

SM-62: Allowed translators to access values modified by previous tran…

ievgenvarava committed:
SM-63: Moved itemExcept logic from mapper configuration into dedicated translator functions

ievgenvarava committed:
Merge remote-tracking branch 'origin/tests' into fix/SM-63-mapper-item-except

ievgenvarava committed:
SM-63: Added unit tests for filter functions.

valerio8787 committed:
SM-61: implemented stream interface concept

valerio8787 committed:
Fixed architecture violations

valerio8787 committed:
Merge branch 'feature/clear-configuration' into feature/SM-61-stream-interface

PhilinTv commented on src/SprykerMiddleware/Service/Process/Model/StreamServiceInterface.php:
```diff
@@ -2,20 +2,32 @@
+     * Specification:
+     * - Read data from given stream.
+     *
+     * @api
```
This interface is not a part of Core API.

PhilinTv commented on src/SprykerMiddleware/Service/Process/ProcessServiceInterface.php:
```diff
@@ -2,20 +2,32 @@
     /**
-     * @param resource $stream
+     * Specification:
+     * - Write given data to given JSON stream.
```
Why are you mixing Stream (read/write) and data formatter everywhere?

PhilinTv commented on src/SprykerMiddleware/Zed/Process/Business/Pipeline/PipelineInterface.php:
```diff
@@ -22,4 +23,21 @@ public function pipe(callable $operation);
     public function process($payload);
+
+    /**
+     * Process the payload.
```
Is that Specification?

PhilinTv commented on src/SprykerMiddleware/Zed/Process/ProcessDependencyProvider.php:
```diff
@@ -208,33 +225,43 @@ protected function getProcesses(): array
+    protected function getStreamPluginsStack()
+    {
+        return [
+            new JsonStreamPlugin(),
```
Should be provided on project level.

ievgenvarava committed:
SM-63: Added whitelist keys assotiative filter, removed item mapping and array filtering from mapper.

ievgenvarava committed:
SM-63: Moved translator function option values to constants.

ievgenvarava committed:
Merge pull request #12 from spryker-middleware/feature/read-map-from-file

Added map reading into abstract map class.

ievgenvarava committed:
SM-63: Removed mapArray() method from mapper, adjusted to use mapKey() from mapByRule() for arrays.

ievgenvarava committed:
Merge pull request #15 from spryker-middleware/fix/SM-63-mapper-item-except

Fix/sm 63 mapper item except

valerio8787 committed:
SM-61: redone Pipeline/Stage interfaces to pass streams in process
method call instead setters

valerio8787 committed:
SM-61:

- got rid from JsonStream Wrapper
- added ReadStreamInterface and WriteStreamInterface interfaces
- removed Aggregator stuff
- refactored Service Layer

valerio8787 committed:
SM-61: Fixed ProcessServiceStream specification

valerio8787 committed:
Merge remote-tracking branch 'remotes/origin/dev' into feature/SM-61-stream-interface

# Conflicts:
#	src/SprykerMiddleware/Zed/Process/Business/Log/Config/MiddlewareLoggerConfig.php
#	src/SprykerMiddleware/Zed/Process/Business/Mapper/Mapper.php
#	src/SprykerMiddleware/Zed/Process/Business/Process/Processor.php

valerio8787 commented:
Already merged
